### PR TITLE
Refactor ONNX AveragePool to align Opsets 7, 10, 11, and 19 (#20553)

### LINKED ISSUE #20553: [Good First Issue]: Align behavior of ONNX Frontend operator AveragePool-7, 10, 11, 19 with original framework
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of AveragePool for next list of opsets: [opset 7](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#7),  [opset 10](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#10),  [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#11),  [opset 19](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#19)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#AveragePool)
More details can be found in ONNX Changelog: [opset 7](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#7),  [opset 10](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#10),  [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#11),  [opset 19](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#19)
1. Operator already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Example Pull Requests

_No response_

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by tani125**
- Split AveragePool implementation into opset_1 and opset_7 namespaces
- opset_1::average_pool calls make_avg_pool() for backward compatibility
- opset_7::average_pool calls make_avg_pool_opset7() for opsets 7+
- Updated ONNX_OP registrations: Opset 1 -> opset_1, Opsets 7,10,11,19 -> opset_7
- Added make_avg_pool_opset7() method to PoolingFactory class
- Implemented explicit handling of count_include_pad (Opset 7), ceil_mode (Opset 10), and dilations (Opset 11)
- Uses !count_include_pad for exclude_pad parameter as required

### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

## Summary
This PR refactors the ONNX AveragePool operator implementation to properly handle different ONNX opset versions by separating the implementation into opset-specific namespaces.

## Related Issue
Closes #20553

### 1. Split AveragePool Implementation (`average_pool.cpp`)
- Created separate `opset_1` and `opset_7` namespaces
- `opset_1::average_pool` → calls `PoolingFactory(node).make_avg_pool()` (backward compatible)
- `opset_7::average_pool` → calls `PoolingFactory(node).make_avg_pool_opset7()` (new implementation)

### 2. Updated ONNX_OP Registrations
- Opset 1 → mapped to `opset_1::average_pool`
- Opsets 7, 10, 11, and 19 → mapped to `opset_7::average_pool`

### 3. Added New Method to PoolingFactory (`pooling_factory.hpp` & `pooling_factory.cpp`)
- Added `make_avg_pool_opset7()` method declaration in `pooling_factory.hpp`
- Implemented `make_avg_pool_opset7()` in `pooling_factory.cpp` with explicit handling of:
  - **count_include_pad** (introduced in Opset 7) - explicitly read and used as `!count_include_pad` for `exclude_pad` parameter
  - **ceil_mode** (introduced in Opset 10) - handled via `m_rounding_type` (already set in constructor)
  - **dilations** (introduced in Opset 11) - handled via `m_dilations` (already set in constructor)

## Technical Details

The new `make_avg_pool_opset7()` method:
- Reads `count_include_pad` attribute explicitly (defaults to 0 if not present)
- Uses `!count_include_pad` for the `exclude_pad` parameter in AvgPool constructor (as required by ONNX spec)
- Selects `v1::AvgPool` when dilations are unit (all 1), otherwise uses `v16::AvgPool`

## Backward Compatibility
✅ Fully backward compatible - Opset 1 models continue to use the original `make_avg_pool()` implementation

## Testing
- [ ] Run C++ tests: `ov_onnx_frontend_tests --gtest_filter=*average_pool*`
- [ ] Run Python tests: `pytest test_ops_convpool.py::test_pool_average`
- [ ] Verify opsets 1, 7, 10, 11, and 19 all work correctly

## Files Changed
- `src/frontends/onnx/frontend/src/op/average_pool.cpp`
- `src/frontends/onnx/frontend/src/utils/pooling_factory.hpp`
- `src/frontends/onnx/frontend/src/utils/pooling_factory.cpp`

### CODE REVIEW

**Review by mvafin — CHANGES_REQUESTED** [review]
Please add tests for `AveragePool` with `count_include_pad` attribute

File: src/frontends/onnx/frontend/src/op/average_pool.cpp [hunk 1]
```diff
@@ -4,18 +4,31 @@
...
+    }
 
+// Registering the different versions
 ONNX_OP("AveragePool", OPSET_SINCE(1), ai_onnx::opset_1::average_pool);
-}  // namespace opset_1
+ONNX_OP("AveragePool", OPSET_SINCE(7), ai_onnx::opset_7::average_pool);
+// We map 10, 11, and 19 to the newer logic as well
+ONNX_OP("AveragePool", OPSET_SINCE(10), ai_onnx::opset_7::average_pool);
+ONNX_OP("AveragePool", OPSET_SINCE(11), ai_onnx::opset_7::average_pool);
+ONNX_OP("AveragePool", OPSET_SINCE(19), ai_onnx::opset_7::average_pool);
```
[the remark this answers is not in the document]
Comment (mvafin) [reply src/frontends/onnx/frontend/src/op/average_pool.cpp:29]: Agree, better use `OPSET_RANGE`
